### PR TITLE
change the load/save methods of maf_id_table

### PR DESCRIPTION
**Pull request by hiroshinoji**
Related to #23. We selected not to serialize json object into .maf_id_table.tsv, but to save object mapping object->ids with pickle, which user can load without maflib dependencies.
The data structure of object in .maf_id_table is changed from dict(Parameter, id) to list(dict), in which dict is converted from Parameter, and index of list is corresponds to the id of that object.

### CODE REVIEW

File: maflib/core.py [hunk 1]
```diff
@@ -476,22 +476,43 @@ def __init__(self, path, text_path):
...
     def save(self):
         """Serializes the table to the file at self.path."""
+
+        parameter_ids = [(param, int(id)) for (param, id) in self._table.items()]
+        parameter_ids.sort(key=lambda param_and_id: param_and_id[1])
+        
         with _create_file(self.path) as f:
-            pickle.dump(self._table, f)
+            # We don't save the self._table, which type is dict(Parameter,int) directly,
+            # instead saves list(dict), which index corresponds to the id of the item.
```
Comment (beam2d) [inline maflib/core.py:493]: This sentence may be improved. Will I fix it after the merge?

File: maflib/core.py [hunk 2]
```diff
@@ -476,22 +476,43 @@ def __init__(self, path, text_path):
...
+            # This is for deserializing parameter->id mappings outside of map, without
+            # maflib and waflib dependencies. Parameter class is defined in maflib,
+            # so user cannot decode original _table object without maflib libraries.
+            # When deserializaing, ``self._table`` is load by ``_load_table``.
+            max_param_id = parameter_ids[-1][1]
```
Comment (beam2d) [inline maflib/core.py:498]: Support the edge case of empty `parameter_ids`.

**Comment by beam2d** [comment]
LGTM except above comments!

File: maflib/core.py (hunk 2 above)
Comment (hiroshinoji) [inline maflib/core.py:498]: change to leave the function first in that case

File: maflib/core.py (hunk 1 above)
Comment (hiroshinoji) [inline maflib/core.py:493]: yes, please.

**Comment by beam2d** [comment]
:+1:

**Pull request merged**